
 This patch should correct the FreeBSD mutex crash that happens in
 the Director after a sudden clock shift or after the "reload" command.
 Apply it to 1.38.10 (will not work with prior version) with:

  cd <bacula-source>
  patch -p0 <1.38.10-scheduler.patch
  make
  make install
  ...



Index: src/dird/scheduler.c
===================================================================
RCS file: /cvsroot/bacula/bacula/src/dird/scheduler.c,v
retrieving revision 1.33.2.2
diff -u -r1.33.2.2 scheduler.c
--- src/dird/scheduler.c	4 Jun 2006 12:24:39 -0000	1.33.2.2
+++ src/dird/scheduler.c	24 Jun 2006 17:25:58 -0000
@@ -151,6 +151,7 @@
               free(next_job);
           }
           schedules_invalidated = false;
+          unlock_jobs();
           goto again;
       }
       unlock_jobs();
